```
FAILED: test/unit/test_rz_test.p/test_rz_test.c.o 
gcc -Itest/unit/test_rz_test.p -I. -I.. -I../librz/include -Itest/unit -I../test/unit -Isubprojects/sdb -I../subprojects/sdb -Isubprojects/sdb/src -I../subprojects/sdb/src -I../shlr/spp -Ishlr -I../shlr -I../librz/asm/arch/include -I../librz/asm/arch -I../librz/asm/arch/h8300 -I../librz/asm/arch/hexagon -I../librz/asm/arch/msp430 -I../librz/asm/arch/rsp -I../librz/asm/arch/mcore -I../librz/asm/arch/v850 -I../librz/asm/arch/propeller -I../librz/asm/arch/ebc -I../librz/asm/arch/cr16 -I../librz/asm/arch/8051 -I../librz/asm/arch/v810 -I../librz/asm/arch/or1k -I../librz/analysis/arch/gb -I../librz/analysis/arch/hexagon -I../librz/bin/format/elf -I../librz/bin/mangling -I../librz/bin/format -I../shlr/heap/include -I../librz/type/parser -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g --std=gnu99 -D_GNU_SOURCE -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -MD -MQ test/unit/test_rz_test.p/test_rz_test.c.o -MF test/unit/test_rz_test.p/test_rz_test.c.o.d -o test/unit/test_rz_test.p/test_rz_test.c.o -c ../test/unit/test_rz_test.c
In file included from ../test/unit/test_rz_test.c:7:
../test/unit/../../binrz/rz-test/run.c: In function ‘rz_test_run_test’:
../test/unit/../../binrz/rz-test/run.c:495:219: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  495 |  if (!success && test->type == RZ_TEST_TYPE_CMD && strstr(test->path, "/dbg") && (!out->out || (!strstr((char *)out->out, "WARNING:") && !strstr((char *)out->out, "ERROR:") && !strstr((char *)out->out, "FATAL:"))) && ((char *)!out->err || (!strstr(out->err, "Sanitizer") && !strstr((char *)out->err, "runtime error:")))) {
      |                                                                                                                                                                                                                           ^
../test/unit/../../binrz/rz-test/run.c:495:252: error: pointer targets in passing argument 1 of ‘strstr’ differ in signedness [-Werror=pointer-sign]
  495 |  if (!success && test->type == RZ_TEST_TYPE_CMD && strstr(test->path, "/dbg") && (!out->out || (!strstr((char *)out->out, "WARNING:") && !strstr((char *)out->out, "ERROR:") && !strstr((char *)out->out, "FATAL:"))) && ((char *)!out->err || (!strstr(out->err, "Sanitizer") && !strstr((char *)out->err, "runtime error:")))) {
      |                                                                                                                                                                                                                                                         ~~~^~~~~
      |                                                                                                                                                                                                                                                            |
      |                                                                                                                                                                                                                                                            unsigned char *
In file included from ../librz/include/rz_types.h:250,
                 from ../librz/include/rz_util.h:7,
                 from ../test/unit/../../binrz/rz-test/rz_test.h:7,
                 from ../test/unit/../../binrz/rz-test/rz-test.c:4,
                 from ../test/unit/test_rz_test.c:5:
/usr/include/string.h:330:14: note: expected ‘const char *’ but argument is of type ‘unsigned char *’
  330 | extern char *strstr (const char *__haystack, const char *__needle)
      |              ^~~~~~
cc1: all warnings being treated as errors
```

--- a/binrz/rz-test/run.c
+++ b/binrz/rz-test/run.c
@@ -492,7 +492,7 @@ RZ_API RzTestResultInfo *rz_test_run_test(RzTestRunConfig *config, RzTest *test)
 #error RZ_ASSERT_STDOUT undefined or 0
 #endif
 	RzSubprocessOutput *out = ret->proc_out;
-	if (!success && test->type == RZ_TEST_TYPE_CMD && strstr(test->path, "/dbg") && (!out->out || (!strstr((char *)out->out, "WARNING:") && !strstr((char *)out->out, "ERROR:") && !strstr((char *)out->out, "FATAL:"))) && ((char *)!out->err || (!strstr((char *)out->err, "Sanitizer") && !strstr((char *)out->err, "runtime error:")))) {
+	if (!success && test->type == RZ_TEST_TYPE_CMD && strstr(test->path, "/dbg") && (!out->out || (!strstr((char *)out->out, "WARNING:") && !strstr((char *)out->out, "ERROR:") && !strstr((char *)out->out, "FATAL:"))) && (!out->err || (!strstr((char *)out->err, "Sanitizer") && !strstr((char *)out->err, "runtime error:")))) {
 		broken = true;
 	}
 #endif

--- a/librz/crypto/p/crypto_aes_algo.c
+++ b/librz/crypto/p/crypto_aes_algo.c
@@ -71,9 +71,9 @@ void aes_expkey(const aes_state_t *st, expkey_t *ek) {
 				tk[i++] ^= tk[j++];
 			}
 			tt = tk[st->columns / 2 - 1];
-			tk[st->columns / 2] ^= Sbox[(ut8)tt] ^ Sbox[(ut8)(tt >> 8)] << 8 ^
-				Sbox[(ut8)(tt >> 16)] << 16 ^
-				Sbox[(ut8)(tt >> 24)] << 24;
+			tk[st->columns / 2] ^= U(Sbox[(ut8)tt]) ^ U(Sbox[(ut8)(tt >> 8)]) << 8 ^
+				U(Sbox[(ut8)(tt >> 16)]) << 16 ^
+				U(Sbox[(ut8)(tt >> 24)]) << 24;
 			for (j = st->columns / 2, i = j + 1; i < st->columns;) {
 				tk[i++] ^= tk[j++];
 			}

--- a/librz/crypto/p/crypto_aes_algo.c
+++ b/librz/crypto/p/crypto_aes_algo.c
@@ -3,9 +3,10 @@
 
 #include "crypto_aes_algo.h"
 
-#define Nb 4 //  number of columns in the state & expanded key
-#define Nr 16 // max number of rounds in encryption
-#define Nk 8 //  max number of columns in a key
+#define Nb   4 //  number of columns in the state & expanded key
+#define Nr   16 // max number of rounds in encryption
+#define Nk   8 //  max number of columns in a key
+#define U(x) ((ut32)(x))
 
 static const ut8 Rcon[30] = {
 	0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80,
@@ -43,10 +44,10 @@ void aes_expkey(const aes_state_t *st, expkey_t *ek) {
 
 	// Copy user material bytes into temporary ints
 	for (i = 0; i < st->columns; i++) {
-		tk[i] = *key++ << 24;
-		tk[i] |= *key++ << 16;
-		tk[i] |= *key++ << 8;
-		tk[i] |= *key++;
+		tk[i] = U(*key++) << 24;
+		tk[i] |= U(*key++) << 16;
+		tk[i] |= U(*key++) << 8;
+		tk[i] |= U(*key++);
 	}
 
 	// Copy values into round key arrays
@@ -58,8 +59,8 @@ void aes_expkey(const aes_state_t *st, expkey_t *ek) {
 	while (t < round_key_count) {
 		// Extrapolate using phi (the round key evolution function)
 		tt = tk[st->columns - 1];
-		tk[0] ^= Sbox[(ut8)(tt >> 16)] << 24 ^ Sbox[(ut8)(tt >> 8)] << 16 ^
-			Sbox[(ut8)tt] << 8 ^ Sbox[(ut8)(tt >> 24)] ^ Rcon[idx++] << 24;
+		tk[0] ^= U(Sbox[(ut8)(tt >> 16)]) << 24 ^ U(Sbox[(ut8)(tt >> 8)]) << 16 ^
+			U(Sbox[(ut8)tt]) << 8 ^ U(Sbox[(ut8)(tt >> 24)]) ^ U(Rcon[idx++]) << 24;
 
 		if (st->columns != 8) {
 			for (i = 1, j = 0; i < st->columns;) {
@@ -106,28 +107,28 @@ void aes_encrypt(aes_state_t *st, ut8 *in, ut8 *result) {
 	ut32 t0, t1, t2, t3, tt;
 	ut32 a0, a1, a2, a3, r;
 
-	t0 = *in++ << 24;
-	t0 |= *in++ << 16;
-	t0 |= *in++ << 8;
-	t0 |= *in++;
+	t0 = U(*in++) << 24;
+	t0 |= U(*in++) << 16;
+	t0 |= U(*in++) << 8;
+	t0 |= U(*in++);
 	t0 ^= ek.key0[0][0];
 
-	t1 = *in++ << 24;
-	t1 |= *in++ << 16;
-	t1 |= *in++ << 8;
-	t1 |= *in++;
+	t1 = U(*in++) << 24;
+	t1 |= U(*in++) << 16;
+	t1 |= U(*in++) << 8;
+	t1 |= U(*in++);
 	t1 ^= ek.key0[0][1];
 
-	t2 = *in++ << 24;
-	t2 |= *in++ << 16;
-	t2 |= *in++ << 8;
-	t2 |= *in++;
+	t2 = U(*in++) << 24;
+	t2 |= U(*in++) << 16;
+	t2 |= U(*in++) << 8;
+	t2 |= U(*in++);
 	t2 ^= ek.key0[0][2];
 
-	t3 = *in++ << 24;
-	t3 |= *in++ << 16;
-	t3 |= *in++ << 8;
-	t3 |= *in++;
+	t3 = U(*in++) << 24;
+	t3 |= U(*in++) << 16;
+	t3 |= U(*in++) << 8;
+	t3 |= U(*in++);
 	t3 ^= ek.key0[0][3];
 
 	// Apply Round Transforms
@@ -185,28 +186,28 @@ void aes_decrypt(aes_state_t *st, ut8 *in, ut8 *result) {
 	ut32 t0, t1, t2, t3, tt;
 	ut32 a0, a1, a2, a3, r;
 
-	t0 = *in++ << 24;
-	t0 |= *in++ << 16;
-	t0 |= *in++ << 8;
-	t0 |= *in++;
+	t0 = U(*in++) << 24;
+	t0 |= U(*in++) << 16;
+	t0 |= U(*in++) << 8;
+	t0 |= U(*in++);
 	t0 ^= ek.key1[0][0];
 
-	t1 = *in++ << 24;
-	t1 |= *in++ << 16;
-	t1 |= *in++ << 8;
-	t1 |= *in++;
+	t1 = U(*in++) << 24;
+	t1 |= U(*in++) << 16;
+	t1 |= U(*in++) << 8;
+	t1 |= U(*in++);
 	t1 ^= ek.key1[0][1];
 
-	t2 = *in++ << 24;
-	t2 |= *in++ << 16;
-	t2 |= *in++ << 8;
-	t2 |= *in++;
+	t2 = U(*in++) << 24;
+	t2 |= U(*in++) << 16;
+	t2 |= U(*in++) << 8;
+	t2 |= U(*in++);
 	t2 ^= ek.key1[0][2];
 
-	t3 = *in++ << 24;
-	t3 |= *in++ << 16;
-	t3 |= *in++ << 8;
-	t3 |= *in++;
+	t3 = U(*in++) << 24;
+	t3 |= U(*in++) << 16;
+	t3 |= U(*in++) << 8;
+	t3 |= U(*in++);
 	t3 ^= ek.key1[0][3];
 
 	// Apply round transforms

